app/workers/application_worker.rb

```diff
@@ -0,0 +1,7 @@
+class ApplicationWorker
+  include Sidekiq::Worker
+
+  def run_with_advisory_lock(model, id, &block)
+    ApplicationRecord.with_advisory_lock("#{model}-#{id}", timeout_seconds: 0) { block.call }
+  end
```

Comment:
Any reason you've gone with using block.call rather than yield? It feels less idiomatic and provides a more cryptic error if a block is missing, though I was wondering if I missed a nuace?

By the way I think this should be a private method